File: Selenium/Activities/Activity22.java
package Activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity22 {
    public static void main(String[] args) throws InterruptedException {
        // Setup the Firefox driver(GeckoDriver)
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        WebDriver driver = new FirefoxDriver();
        Actions builder = new Actions(driver);
        driver.get("https://v1.training-support.net/selenium/popups");
        System.out.println("Title of the page: " +driver.getTitle());
        // Find the Sign-in button and hover over it
        WebElement button = driver.findElement(By.className("orange"));
        builder.moveToElement(button).build().perform();
        // Print the tooltip message
        String toolTip = button.getAttribute("data-tooltip");
        System.out.println("Tooltip message: " +toolTip);
        // Click the button to open the Sign-in form
        button.click();
        // Enter the credentials
        driver.findElement(By.id("username")).sendKeys("admin");
        driver.findElement(By.name("Password")).sendKeys("password");
        driver.findElement(By.xpath("//button[text()='Log in']")).click();
        Thread.sleep(2000);
        String message = driver.findElement(By.id("action-confirmation")).getText();
        System.out.println("Login message: " +message);
        // Close the browser
        driver.quit();
    }
}

File: Selenium/Activities/Activity22.py
import time

from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Open the browser to the URL
    driver.get("https://v1.training-support.net/selenium/popups")
    print("Title of the page is: ", driver.title)
    # Find the Sign-in button and hover over it
    button = driver.find_element(By.CLASS_NAME, "orange")
    # Set up actions object
    actions = webdriver.ActionChains(driver)
    actions.move_to_element(button).perform()
    # Print the tooltip message
    toolTip = button.get_attribute("data-tooltip")
    print("Tooltip message: ", toolTip)
    # Click the button to open the Sign-in form
    button.click()
    # Enter the credentials and click login
    driver.find_element(By.NAME, "Username").send_keys("admin")
    driver.find_element(By.ID, "password").send_keys("password")
    driver.find_element(By.XPATH, "//button[@onclick='signIn()']").click()
    time.sleep(2)
    confirm_Msg = driver.find_element(By.ID, "action-confirmation").text
    print("Login confirmation message: ", confirm_Msg)

